astropy/modeling/core.py

```diff
@@ -2416,24 +2417,82 @@ def render_model(model, arr=None, coords=None, copy=True):
     ----------
     model : `Model`
         Model to be evaluated.
-    arr : `numpy.ndarray`, optional
-        Array on which the model is evaluated.
+    arr : `numpy.ndarray` 
+        Array on which the model is evaluated. 
     coords : array-like, optional
-        Coordinate arrays mapping to ``array``, such that 
+        Coordinate arrays mapping to ``arr``, such that 
         ``arr[coords] == arr``.
+    wcs : `~astropy.wcs.WCS`, optional
+        A WCS object mapping to the input ``arr``. 
+    origin: int, optional
+        The coordinate in the upper left corner of the image. In FITS and 
+        Fortran standards, this is 1. In Numpy and C standards this is 0. 
     copy : bool, optional
         Default is ``True``. If ``False``, ``arr`` will be modified in place. 
+    return_masked : bool, optional
```

Comment:
What is this? Is this really needed?
It complicates the function quite a bit ... generally you should try to avoid writing functions that sometimes return one thing and sometimes another thing or multiple things.